app/controllers/chores_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ChoresController < ApplicationController
+  def create
+    begin
+      chore = Chore.create(chore_params)
+      render status: 201,  json: chore.to_json
+    rescue
+      render status: 400
+    end
+  end
+
+  def destroy
+    begin
+      params.require(:id)
+      Chore.destroy_by(id: params[:id])
+      render status: 200
+    rescue
+      render status: 400
+    end
+  end
+
+  def update
+    begin
+      id = params[:id]  
+      chore = Chore.find_by(id: id)    
+      chore.update(chore_params)
+      render status: 200
+    rescue
+      render status: 400
+    end
+  end
+
+  private
+
+  def chore_params
+    params.require(:chore).permit(:title, :description, :hideout_id, :assignee_id, :due_date)
```

Comment:
<img width="459" alt="Screenshot 2023-06-03 at 9 19 37 PM" src="https://github.com/tryhideout/web/assets/44104695/1532a429-df58-4bbc-84fa-c83a5610685d">
Here's the issue with this type of function. When you send a request without the hideout_id or description, it'll give a MySQL error, which is the last thing we want. We should be catching any "bad requests" before it gets to the database. Right now, all your methods are going to throw `500 Internal Server Error` which your `begin-rescue` block returns as a `400 Bad Request`.

I would say to stick with individualized `params.require` calls per function since there's only like 3 and each of them have slightly different requirements. It's not a case of code duplication when it's happening like twice and when some of them have different inputs.